src/ProgressOnderwijsUtils/Pocos/BulkInsertTarget.cs

```diff
@@ -40,10 +47,59 @@ public BulkInsertTarget With(BulkCopyFieldMappingMode mode)
         public BulkInsertTarget With(SqlBulkCopyOptions options)
             => new BulkInsertTarget(TableName, Columns, Mode, options);
 
+        [SuppressMessage("ReSharper", "PossibleMultipleEnumeration")]
+        static (T[] head, IEnumerable<T> fullSequence) PeekAtPrefix<T>(IEnumerable<T> enumerable, int firstN)
+        {
+            var enumerator = enumerable.GetEnumerator();
+            var buffer = new List<T>();
+            for (var i = 0; i < firstN; i++) {
+                if (enumerator.MoveNext()) {
+                    buffer.Add(enumerator.Current);
+                } else {
+                    var partialHead = buffer.ToArray();
+                    return (partialHead, partialHead);
+                }
+            }
+            var completeHead = buffer.ToArray();
+            if (enumerable is IReadOnlyList<T> || enumerable is ICollection<T>) {
+                return (completeHead, enumerable);
+            }
+
+            IEnumerable<T> FullSequence()
+            {
+                if (completeHead.PretendNullable() != null) {
+                    foreach (var item in completeHead) {
+                        yield return item;
+                    }
+#pragma warning disable IDE0059 // Unnecessary assignment of a value
+                    completeHead = null;
+#pragma warning restore IDE0059 // Unnecessary assignment of a value
+                } else {
+                    enumerator = enumerable.GetEnumerator();
+                }
+
+                while (enumerator.MoveNext()) {
+                    yield return enumerator.Current;
+                }
+                enumerator.Dispose();
+            }
+
+            return (completeHead, FullSequence());
+        }
+
         public void BulkInsert<[MeansImplicitUse(ImplicitUseKindFlags.Access, ImplicitUseTargetFlags.WithMembers)]
             T>(SqlConnection sqlConn, IEnumerable<T> pocos, CommandTimeout timeout = default, CancellationToken cancellationToken = default)
             where T : IReadImplicitly
         {
+            if (Options == defaultBulkCopyOptions) {
+                var (head, all) = PeekAtPrefix(pocos, 10);

```

Comment:
die 10 naar const of var?

Reply:
Ik denk dat ik em ook ga tweaken naar iets lagers, maar idd, een const lijkt me goed.